### PR TITLE
ramips: mt7620: Adding support for Dlink DAP-1620-A2

### PR DESCRIPTION
**Pull request by sendurai**
Hello,

Adding support for D-Link DAP-1620-A2 device. 

A bit older device, but nevertheless has a decent hardware (64/8 and both 2.4 & 5). The firmware package needs to be pushed to device through U-Boot serial port and can't be done through Web UI, since the recovery occupies the second part of NAND. Can add the details to the wiki page once the code is merged and compiled.

Waiting for OpenWrt based compile of image. Anyways added the OpenWrt boot log to the wiki already. Available here:
https://openwrt.org/inbox/toh/d-link/d-link_dap-1620

Made few updates based on previous comment as noted below. Any further questions, please let me know.
https://github.com/openwrt/openwrt/pull/16911

Thanks

### CODE REVIEW

File: target/linux/ramips/dts/mt7620a_dlink_dap-1620-a2.dts [hunk 1]
```diff
@@ -0,0 +1,203 @@
+#include "mt7620a.dtsi"
```
Comment (musashino205) [inline target/linux/ramips/dts/mt7620a_dlink_dap-1620-a2.dts:1]: Please insert `// SPDX-License-Identifier: ` with identifier(s) of your preferred license.

https://spdx.org/licenses/

File: target/linux/ramips/dts/mt7620a_dlink_dap-1620-a2.dts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "nvram";
+				reg = <0x30000 0x10000>;
+			};
+
+			factory: partition@40000 {
```
Comment (musashino205) [inline target/linux/ramips/dts/mt7620a_dlink_dap-1620-a2.dts:100]: Please remove this unused DT label (`factory: `).

File: target/linux/ramips/dts/mt7620a_dlink_dap-1620-a2.dts [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+					macaddr_factory_ffe8: macaddr@ffe8 {
+						compatible = "mac-base";
+						reg = <0xffe8 0x11>;
+						#nvmem-cell-cells = <1>;
+					};
+				};
+			};
+
+			firmware: partition@50000 {
```
Comment (musashino205) [inline target/linux/ramips/dts/mt7620a_dlink_dap-1620-a2.dts:132]: ditto (`firmware: `)

File: target/linux/ramips/dts/mt7620a_dlink_dap-1620-a2.dts [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+};
+
+&pcie0 {
+	wifi@0,0 {
+		reg = <0x0000 0 0 0 0>;
+		nvmem-cells = <&eeprom_factory_8000>, <&macaddr_factory_ffd4 0>;
+		nvmem-cell-names = "eeprom", "mac-address";
+		ieee80211-freq-limit = <5000000 6000000>;
+	};
+};
```
Comment (musashino205) [inline target/linux/ramips/dts/mt7620a_dlink_dap-1620-a2.dts:203]: Please put a newline at the end of file.

**Comment by raenye** [comment]
Did you manage to fix the RTL8211E gigabit ethernet problem?
See forum thread: https://forum.openwrt.org/t/support-for-dlink-dap-1620/72667/111

Also, you need to squash all your commits into a single commit. Never use `git merge`, only `git rebase`.

**Comment by sendurai** [comment]
> Did you manage to fix the RTL8211E gigabit ethernet problem? See forum thread: https://forum.openwrt.org/t/support-for-dlink-dap-1620/72667/111
> 
> Also, you need to squash all your commits into a single commit. Never use `git merge`, only `git rebase`.

Just to highlight this is Revision A hardware. (A1/A2). (mt7620). The discussion thread seems to focus on B hardware?

Yes, the 1G speed works. Luci image attached. 
DAP-1620-A2-Switch

**Comment by sendurai** [comment]
Commits merged into single commit and ready for review/merge. Thanks

**Comment by musashino205** [comment]
Please use `Add` instead of `Adding` for the commit title.

https://openwrt.org/submitting-patches#submission_guidelines

> - write commit subject and message in the imperative form: “add support for X”, NOT “added support for X”

**Comment by raenye** [comment]
> The discussion thread seems to focus on B hardware?

Actually no, the vast majority of the discussion is about rev A. Only the last couple of posts are about rev B.
Perhaps you should quickly go over the thread and see whether anything there is still relevant.

**Comment by sendurai** [comment]
Sorry. Will reopen another pull request. Had to delete my local branch and cannot reopen another request here.

**Comment by Neustradamus** [comment]
Replaces:
- https://github.com/openwrt/openwrt/pull/16911

Replaced by:
- https://github.com/openwrt/openwrt/pull/17018
- https://github.com/openwrt/openwrt/pull/17020
- https://github.com/openwrt/openwrt/pull/17027

**Comment by sendurai** [comment]
Since I deleted that branch and created one was not able to re-open. This is available in a new PR.
https://github.com/openwrt/openwrt/pull/17027

**Pull request closed without merging**